packages/core/src/stateUtils.ts

```diff
@@ -96,6 +91,22 @@ export function getChildren<TC, TE extends EventObject>(
   return keys(stateNode.states).map(key => stateNode.states[key]);
 }
 
+export function getProperAncestors<TContext, TEvent extends EventObject>(
+  stateNode: StateNode<TContext, any, TEvent>,
+  toStateNode: StateNode<TContext, any, TEvent> | null
+): Array<typeof stateNode> {
+  const ancestors = new Set<typeof stateNode>();
```

Comment:
this is a fresh Set, since we only add ancestors recursively to it there is no chance for duplicates being added into that, so instead of creating a Set upfront and casting it to an Array at the end we should be able to just create an Array and push to it